worlds/marioland2/__init__.py

```diff
@@ -0,0 +1,576 @@
+                    if coin_name in item_counts:
+                        amount_to_remove = min(-items_to_add, item_counts[coin_name])
+                        item_counts[coin_name] -= amount_to_remove
+                        items_to_add += amount_to_remove
+                        if items_to_add >= 0:
+                            break
+
+            double_progression_items = ["Tree Zone Progression", "Macro Zone Progression", "Pumpkin Zone Progression",
+                                        "Mario Zone Progression", "Turtle Zone Progression"]
+            self.random.shuffle(double_progression_items)
+            while sum(item_counts.values()) > location_count:
+                if double_progression_items:
+                    double_progression_item = double_progression_items.pop()
+                    item_counts[double_progression_item] -= 2
+                    item_counts[double_progression_item + " x2"] = 1
+                    continue
+                if self.options.auto_scroll_mode in ("level_trap_items", "level_cancel_items",
+                                                     "chaos"):
+                    auto_scroll_item = self.random.choice([item for item in item_counts if "Auto Scroll" in item])
+                    level = auto_scroll_item.split("- ")[1]
+                    self.auto_scroll_levels[level_name_to_id[level]] = 0
+                    del item_counts[auto_scroll_item]
+                    continue
+                raise Exception(f"Too many items in the item pool for Super Mario Land 2 player {self.multiworld.player_name[self.player]}")
+                # item = self.random.choice(list(item_counts))
+                # item_counts[item] -= 1
+                # if item_counts[item] == 0:
+                #     del item_counts[item]
+                # self.multiworld.push_precollected(self.create_item(item))
+
+        self.coin_fragments_required = max((item_counts["Mario Coin Fragment"]
+                                           * self.options.mario_coin_fragments_required_percentage) // 100, 1)
+
+        for item_name, count in item_counts.items():
+            self.multiworld.itempool += [self.create_item(item_name) for _ in range(count)]
+
+    def fill_slot_data(self):
+        return {
+            "energy_link": self.options.energy_link.value
+        }
+
+    def create_item(self, name: str) -> Item:
+        return MarioLand2Item(name, items[name], self.item_name_to_id[name], self.player)
+
+    def get_filler_item_name(self):
+        return "1 Coin"
+
+    def modify_multidata(self, multidata: dict):
+        rom_name = bytearray(f'AP{Utils.__version__.replace(".", "")[0:3]}_{self.player}_{self.multiworld.seed:11}\0',
+                             'utf8')[:21]
+        rom_name.extend([0] * (21 - len(rom_name)))
+        new_name = base64.b64encode(bytes(rom_name)).decode()
+        multidata["connect_names"][new_name] = multidata["connect_names"][self.multiworld.player_name[self.player]]
```

Comment:
You can shorten this
```suggestion
        multidata["connect_names"][new_name] = multidata["connect_names"][self.player_name]
```

Reply:
changed